File: Python/0169_majority_element/solution.py
def majority_element(nums):
    num_count = {}
    length = len(nums)
    for num in nums:
        num_count[num] = num_count.get(num , 0) + 1
        if num_count[num] > length // 2:
            return num
    return None

nums = [2,2,1,1,1,2,2]
print(majority_element(nums))

File: JavaScript/0169_majority_element/solution.js
function majority_element(nums) {
    let num_count = {};
    let len = nums.length;
    for (let num of nums) {
        if (num in num_count) {
            num_count[num] ++;
        } else {
            num_count[num] = 1;
        }
        if (num_count[num] > Math.floor(len / 2)) {
            return num;
        }
    }
    return null;
}

const nums = [2,2,1,1,1,2,2];
console.log(majority_element(nums));